query/src/lib.rs

```diff
@@ -50,11 +46,12 @@ pub enum SrcVar {
 }
 
 /// These are the scalar values representable in EDN.
-#[derive(Clone,Debug,Eq,PartialEq)]
+// Floats and doubles only implement PartialEq, but not Eq.
+#[derive(Clone,Debug,PartialEq)]
 pub enum NonIntegerConstant {
     Boolean(bool),
-    BigInteger(BigInt),
-    Float(OrderedFloat<f64>),
+    BigInteger(String),
```

Comment:
So you want to have u/i64 as source for BigInteger? How should the sign be handled?